src/mol-io/reader/sdf/parser.ts

```diff
@@ -71,9 +72,17 @@ function handleMolFile(tokenizer: Tokenizer) {
     const program = Tokenizer.readLine(tokenizer).trim();
     const comment = Tokenizer.readLine(tokenizer).trim();
 
-    const counts = Tokenizer.readLine(tokenizer);
+    const countsAndVersion = Tokenizer.readLine(tokenizer);
+    const molIsV3 = isV3(countsAndVersion);
 
-    const atomCount = +counts.substr(0, 3), bondCount = +counts.substr(3, 3);
+    let atomCount = NaN;
+    let bondCount = NaN;
+
+    if (molIsV3) {
+        ({ atomCount, bondCount } = handleCountsV3(tokenizer));
+    } else {
+        atomCount = +countsAndVersion.substr(0, 3), bondCount = +countsAndVersion.substr(3, 3);
+    }
```

Comment:
This parts feels a bit weird now. Please add ``handleCountsV2`` and do

```ts
const { atomCount, bondCount } = molIsV3 
    ? handleCountsV3(tokenizer)
    : handleCountsV2(countsAndVersion);
```

Reply:
Good point, your suggestion is much cleaner. I have pushed an update.